tests/api2/test_dataset_recordsize_choices.py

```diff
@@ -0,0 +1,14 @@
+import sys
+import os
+sys.path.append(os.getcwd())
+
+from middlewared.test.integration.utils import call
+
+EXPECTED = set(['512B', '1K', '2K', '4K', '8K', '16K', '32K', '64K', '128K', '256K', '512K', '1M'])
+
+
+def test_pool_dataset_recordsize_choices():
+    with open('/sys/module/zfs/parameters/zfs_max_recordsize', 'w') as f:
```

Comment:
Now almost LGTM, but please fix this https://github.com/truenas/middleware/pull/8739#discussion_r845582528

Reply:
This isn't a unit test....

Reply:
Same issue: it will affect other tests executed in the same environment.

Either convert it back to unit test and mock `open` method, or move retrieving `zfs_max_recordsize` to a plugin method and mock it using our integration test mocking tools.

Reply:
It's using ssh() to make the change

Reply:
So that machine's zfs module parameters would be changed, which will affect other tests executed on the same machine.

Reply:
I'm tired of wasting my time so I've removed the tests.

Reply:
Should I add the test then?

Reply:
sure